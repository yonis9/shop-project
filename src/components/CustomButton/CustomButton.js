import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleButton, ...otherProps}) => (
    <button className={`${isGoogleButton ? 'google-sign-in' : ''} custom-button`} {...otherProps}>
        {children}
    </button>
)

export default CustomButton;