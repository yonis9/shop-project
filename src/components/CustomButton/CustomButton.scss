.custom-button {
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 35px 0 35px;
    font-size: 15px;
    background-color: black;
    color: white;
    text-transform: uppercase;
    font-family: 'Open Sans', sans-serif;
    font-weight: bolder;
    border: none;
    cursor: pointer;
    transition: 0.3s;
  
    &:hover {
      background-color: white;
      color: black;
      border: 1px solid black;
      transition: 0.3s;
    }
  }
  