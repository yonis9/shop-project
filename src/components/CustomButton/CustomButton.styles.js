import styled, { css } from 'styled-components';

const buttonStyles = css`
background-color: black;
color: white;
border: none;

&:hover {
    background-color: white;
    color: black;
    border: 1px solid black;
    }
`

const googleSignInStyles = css`
    background-color: #4285f4;
    color: white;

    &:hover {
        background-color: #326abc;
        border: none;
    }
`

const invertedStyles = css`
    background-color: white;
    color: black;
    border: 1px solid black;
    

    &:hover {
    background-color: black;
    color: white;
    border: none;
    }
`

const getButtonStyles = props => {
    if (props.isGoogleButton) {
        return googleSignInStyles;
    }

    return props.inverted ? invertedStyles : buttonStyles
}

export const CustomButtonContainer = styled.button`
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 25px 0 25px;
    font-size: 15px;
    text-transform: uppercase;
    font-family: 'Open Sans', sans-serif;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: center;
    transition: 0.3s;

    ${getButtonStyles}

    &:focus {
    outline: 0
    }

    &:hover {
        transition: 0.3s;
    }

`