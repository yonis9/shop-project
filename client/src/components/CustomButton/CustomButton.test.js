import React from 'react';
import { shallow } from 'enzyme';

import CustomButton from './CustomButton';

it('should rendered CustomButton component', () => {
    expect(shallow(<CustomButton />)).toMatchSnapshot();
});