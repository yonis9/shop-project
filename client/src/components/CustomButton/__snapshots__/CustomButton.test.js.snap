// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered CustomButton component 1`] = `ShallowWrapper {}`;
